src/scripting/lua_widget_methods.cpp

```diff
@@ -233,14 +233,15 @@ namespace
 	// The caller still have to subtract 1 when passing the index to c++
 	// (0-based) functions
 	template<typename TWidget>
-	void fix_index(lua_State* L, int arg, TWidget& w, bool for_insertion, int& index)
+	void check_index(lua_State* L, int arg, TWidget& w, bool for_insertion, int& index)
 	{
 		int nitems = number_of_items(w);
 		// index == nitems + 1 -> insert at the end.
 		int max = for_insertion ? nitems + 1 : nitems;
 
-		// 0 means insert at end.
+		// 0 means the user didnt specify a value, so insert at end.
 		if(index == 0) {
+			assert(for_insertion);
```

Comment:
Don't use assert for something like this.

Reply:
This condition is checked by the code that calls this function

Reply:
That might be true _now_, but it's not guaranteed by the structure of the function, so someone later could easily come along and change something and expose this assert.

It should be fine to just use `lua_error` or one of the related functions. (Note: if using `lua_error` rather than one of the related functions, you need to first push the error message onto the stack. Using one of the related functions is probably preferred if possible, but I'm not sure.) Alternatively, throwing an exception would be acceptable.